lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -100,29 +104,36 @@ def get_course_topics(self, user=None):
         Get course topics for self.course, using the given user or self.user if
         not provided, and generating absolute URIs with a test scheme/host.
         """
-        return get_course_topics(self.course.id, user or self.user)
+        if user:
+            self.request.user = user
```

Comment:
I think it's a little bit weird to have this function modify self.request. I think it would be better to either change the function interface to not accept the user argument (in which case the caller would have to set self.request.user) or to create a request with the given user in this function.